asyncpg/protocol/codecs/datetime.pyx

```diff
@@ -232,31 +232,361 @@ cdef timetz_decode(ConnectionSettings settings, FastReadBuffer buf):
+            carry, hours = divmod(hours + carry, 24)
+            days += carry
+            time = (hours * 60 * 60 + minutes * 60 + seconds) * 1000000 + microseconds
+
+        self.months = months
+        self.days = days
+        self.time = time
+
+    @property
+    def hours(self):
+        return (self.time // (60 * 60 * 1000000))
+
+    @property
+    def minutes(self):
+        return (self.time // (60 * 1000000)) % 60
+
+    @property
+    def seconds(self):
+        return (self.time // 1000000) % 60
+
+    @property
+    def microseconds(self):
+        return self.time % 1000000
+
+    def __repr__(self):
+        return "Interval(months=%d, days=%d, time=%d)" % (self.months, self.days, self.time)
+
+    def __str__(self):
+        cdef:
+            int32_t hours
+            int32_t mins
+            int32_t secs
+            int32_t usecs
+            int64_t time
+
+        parts = []
+
+        if self.months:
+            parts.append("%d month%s" % (self.months, '' if abs(self.months) == 1 else 's'))
+
+        if self.days:
+            parts.append("%d day%s" % (self.days, '' if abs(self.days) == 1 else 's'))
+
+        if self.time:
+            time, usecs = divmod(self.time, 1000000)
+            time, secs = divmod(time, 60)
+            time, mins = divmod(time, 60)
+            time, hours = divmod(time, 60)
+            sign = '-' if time < 0 else ''
+            if usecs:
+                parts.append("%s%02d:%02d:%02d.%06d" % (sign, hours, mins, secs, usecs))
+            else:
+                parts.append("%s%02d:%02d:%02d" % (sign, hours, mins, secs))
+
+        return ', '.join(parts)
+
+    def __hash__(self):
+        return hash((self.months, self.days, self.time))
+
+    def __richcmp__(self, other, int op):
+        if isinstance(other, Interval):
+            if op == 2: # ==
+                return (self.months == other.months
+                        and self.days == other.days
+                        and self.time == other.time)
+            elif op == 3: # !=
+                return (self.months != other.months
+                        or self.days != other.days
+                        or self.time != other.time)
+            elif op == 0: # <
+                return ((self.months, self.days, self.time)
+                        <
+                        (other.months, other.days, other.time))
+            elif op == 1: # <=
+                return ((self.months, self.days, self.time)
+                        <=
+                        (other.months, other.days, other.time))
+            elif op == 4: # >
+                return ((self.months, self.days, self.time)
+                        >
+                        (other.months, other.days, other.time))
+            elif op == 5: # >=
+                return ((self.months, self.days, self.time)
+                        >=
+                        (other.months, other.days, other.time))
+        return False
```

Comment:
Here too, from my understanding and tests of the Cython __richcmp__(), if `other` is not an instance of `Interval` then it should return `False`. I will try to find again the examples that lead me there.